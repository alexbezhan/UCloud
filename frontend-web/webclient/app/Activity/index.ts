import { ActivityReduxObject } from "DefaultObjects";
import { ScrollRequest } from "Scroll";

export enum ActivityType {
    DOWNLOAD = "download",
    UPDATED = "updated",
    DELETED = "deleted",
    FAVORITE = "favorite",
    INSPECTED = "inspected",
    MOVED = "moved"
}

export interface ActivityGroup {
    type: ActivityType;
    newestTimestamp: number;
    numberOfHiddenResults: number | null;
    items: Activity[];
}

export interface Activity {
    type: ActivityType;
    timestamp: number;
    fileId: string;
    username: string;
    originalFilePath: string;
}

export interface ActivityFilter {
    collapseAt?: number;
    type?: ActivityType;
    minTimestamp?: Date;
    maxTimestamp?: Date;
}

export interface FavoriteActivity extends Activity {
    favorite: boolean;
}

export interface MovedActivity extends Activity {
    newName: string;
}


export interface ActivityDispatchProps {
    onMount: () => void;
    fetchActivity: (scroll: ScrollRequest<number>, filter?: ActivityFilter) => void;
    resetActivity: () => void;
    setRefresh: (refresh?: () => void) => void;
    updateFilter: (filter: Partial<ActivityFilter>) => void;
}

export interface ActivityOwnProps {
}

export type ActivityProps = ActivityReduxObject & ActivityDispatchProps & ActivityOwnProps;
