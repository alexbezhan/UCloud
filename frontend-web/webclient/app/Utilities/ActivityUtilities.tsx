import {ActivityFilter} from "Activity";
import {ScrollRequest} from "Scroll";
import {buildQueryString} from "./URIUtilities";

interface ParamsType extends Omit<ActivityFilter, "maxTimestamp" | "minTimestamp">, Partial<ScrollRequest<number>> {
    maxTimestamp?: number;
    minTimestamp?: number;
}

export const activityQuery = (scroll: ScrollRequest<number>, filter?: ActivityFilter) => {
    const params: ParamsType = {};
    if (!!scroll.offset) params.offset = scroll.offset;
    params.scrollSize = scroll.scrollSize;

    if (filter !== undefined) {
        if (filter.type !== undefined) params.type = filter.type;
        if (filter.collapseAt !== undefined) params.collapseAt = filter.collapseAt;
        if (filter.maxTimestamp !== undefined) params.maxTimestamp = filter.maxTimestamp.getTime();
        if (filter.minTimestamp !== undefined) params.minTimestamp = filter.minTimestamp.getTime();
        if (filter.user !== undefined) params.user = filter.user;
    }

    return buildQueryString("/activity/browse", params);
};

export const activityStreamByPath = (path: string) =>
    `/activity/by-path?path=${encodeURIComponent(path)}`;
