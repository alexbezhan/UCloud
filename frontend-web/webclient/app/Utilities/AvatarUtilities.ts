export const saveAvatarQuery = "/avatar/update";
export const findAvatarQuery = "/avatar/find";
