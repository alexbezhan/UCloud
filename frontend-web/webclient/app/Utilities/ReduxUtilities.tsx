import activity from "Activity/Redux/ActivityReducer";
import * as AppRedux from "Applications/Redux";
import analyses from "Applications/Redux/AnalysesReducer";
import applications from "Applications/Redux/BrowseReducer";
import detailedApplicationSearch from "Applications/Redux/DetailedApplicationSearchReducer";
import * as TaskRedux from "BackgroundTasks/redux";
import dashboard from "Dashboard/Redux/DashboardReducer";
import {initObject} from "DefaultObjects";
import detailedFileSearch from "Files/Redux/DetailedFileSearchReducer";
import fileInfo from "Files/Redux/FileInfoReducer";
import header, {CONTEXT_SWITCH, USER_LOGIN, USER_LOGOUT} from "Navigation/Redux/HeaderReducer";
import sidebar from "Navigation/Redux/SidebarReducer";
import status from "Navigation/Redux/StatusReducer";
import notifications from "Notifications/Redux/NotificationsReducer";
import * as ProjectRedux from "Project/Redux";
import {Action, AnyAction, combineReducers, createStore, Store} from "redux";
import {composeWithDevTools} from "redux-devtools-extension";
import {createResponsiveStateReducer, responsiveStoreEnhancer} from "redux-responsive";
import simpleSearch from "Search/Redux/SearchReducer";
import {responsiveBP} from "ui-components/theme";
import uploader from "Uploader/Redux/UploaderReducer";
import avatar from "UserSettings/Redux/AvataaarReducer";
import hookStore from "Utilities/ReduxHooks";

export function configureStore(
    initialObject: ReduxObject,
    reducers,
    enhancers?
): Store<ReduxObject, AnyAction> {
    const combinedReducers = combineReducers<ReduxObject, AnyAction>(reducers);
    const rootReducer = (state: ReduxObject, action: Action): ReduxObject => {
        if ([USER_LOGIN, USER_LOGOUT, CONTEXT_SWITCH].some(it => it === action.type)) {
            state = initObject();
        }
        return combinedReducers(state, action);
    };
    return createStore<ReduxObject, AnyAction, {}, {}>(rootReducer, initialObject, composeWithDevTools(enhancers));
}

export const responsive = createResponsiveStateReducer(
    responsiveBP,
    {infinity: "xxl"}
);

export const store = configureStore(initObject(), {
    activity,
    dashboard,
    analyses,
    applications,
    header,
    status,
    sidebar,
    uploader,
    notifications,
    simpleSearch,
    detailedFileSearch,
    detailedApplicationSearch,
    fileInfo,
    hookStore,
    ...AppRedux.reducers,
    avatar,
    loading,
    tasks: TaskRedux.reducer,
    project: ProjectRedux.reducer,
    responsive: createResponsiveStateReducer(
        responsiveBP,
        {infinity: "xxl"}),
}, responsiveStoreEnhancer);

function loading(state = false, action: {type: string}): boolean {
    switch (action.type) {
        case "LOADING_START":
            return true;
        case "LOADING_END":
            return false;
        default:
            return state;
    }
}
