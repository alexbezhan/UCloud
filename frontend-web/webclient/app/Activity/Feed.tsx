import * as Module from "Activity";
import {Cloud} from "Authentication/SDUCloudObject";
import {format, formatDistanceToNow} from "date-fns/esm";
import * as React from "react";
import {Link as ReactRouterLink} from "react-router-dom";
import styled from "styled-components";
import {Box, Flex, Text} from "ui-components";
import Icon, {IconName} from "ui-components/Icon";
import Table, {TableCell, TableHeader, TableHeaderCell, TableRow} from "ui-components/Table";
import {colors} from "ui-components/theme";
import {fileInfoPage, getFilenameFromPath, replaceHomeFolder} from "Utilities/FileUtilities";

export function ActivityFeedFrame(props: {containerRef?: React.RefObject<any>, children?: JSX.Element[]}) {
    return (
        <Table>
            <TableHeader>
                <TFRow>
                    <TableHeaderCell width="12em" />
                    <TableHeaderCell width="10.5em" />
                    <TableHeaderCell width="99%" />
                </TFRow>
            </TableHeader>
            <tbody ref={props.containerRef}>
                {props.children}
            </tbody>
        </Table >
    );
}

export const ActivityFeed = ({activity}: {activity: Module.Activity[]}) => (
    <ActivityFeedFrame>
        {groupActivity(activity).map((a, i) => <ActivityFeedItem key={i} activity={a} />)}
    </ActivityFeedFrame>
);

// Performance note: Don't use styled components here.
const ActivityEvent: React.FunctionComponent<{event: Module.Activity}> = props => (
    <div>
        <b>
            <ReactRouterLink to={fileInfoPage(props.event.originalFilePath)}>
                <div className="ellipsis">
                    <Text color="black">{getFilenameFromPath(props.event.originalFilePath)}</Text>
                </div>
            </ReactRouterLink>
        </b>
        {" "}
        <OperationText event={props.event} />
    </div>
);

// Performance note: Don't use styled components here.
const OperationText: React.FunctionComponent<{event: Module.Activity}> = props => {
    switch (props.event.type) {
        case Module.ActivityType.MOVED: {
            return (
                <span>
                    was moved to
                {" "}
                    <b>
                        <ReactRouterLink to={fileInfoPage((props.event as Module.MovedActivity).newName)}>
                            <div className="ellipsis">
                                <Text color="black">
                                    {replaceHomeFolder((props.event as Module.MovedActivity).newName, Cloud.homeFolder)}
                                </Text>
                            </div>
                        </ReactRouterLink>
                    </b>
                </span>
            );
        }

        case Module.ActivityType.FAVORITE: {
            const isFavorite = (props.event as Module.FavoriteActivity).favorite;
            if (isFavorite) {
                return <span>was <b>added to favorites</b></span>;
            } else {
                return <span>was <b>removed from favorites</b></span>;
            }
        }

        default: {
            return <span>was <b>{operationToPastTense(props.event.type)}</b></span>;
        }
    }
};

export const ActivityFeedSpacer = (props: {height: number}) => (
    <tr style={{height: `${props.height}px`}} />
);

interface ActivityFeedProps {
    activity: Module.ActivityGroup;
}

export class ActivityFeedItem extends React.Component<ActivityFeedProps> {
    public shouldComponentUpdate(nextProps: ActivityFeedProps) {
        return this.props.activity.newestTimestamp !== nextProps.activity.newestTimestamp;
    }

    public render() {
        const {activity} = this.props;
        return (
            <TFRow>
                <TableCell>
                    <Text fontSize={1} color="text">
                        {formatDistanceToNow(new Date(activity.newestTimestamp))}
                        <br />
                        {format(new Date(activity.newestTimestamp), "d LLL yyyy HH:mm")}
                    </Text>
                </TableCell>
                <TableCell>
                    <Flex>
                        <Icon mr="0.5em" name={eventIcon(activity.type).icon} />
                        <Text fontSize={2}>{`Files ${operationToPastTense(activity.type)}`}</Text>
                    </Flex>
                </TableCell>
                <TableCell>
                    {activity.items.map((item, idx) =>
                        <ActivityEvent key={idx} event={item} />
                    )}

                    {!!activity.numberOfHiddenResults ?
                        <Box mt={16}>
                            <Text bold>{activity.numberOfHiddenResults} similar results were hidden</Text>
                        </Box>
                        :
                        null
                    }
                </TableCell>
            </TFRow>
        );
    }
}

const operationToPastTense = (operation: Module.ActivityType): string => {
    switch (operation) {
        case Module.ActivityType.DELETED:
            return "deleted";
        case Module.ActivityType.DOWNLOAD:
            return "downloaded";
        case Module.ActivityType.FAVORITE:
            return "favorited";
        case Module.ActivityType.INSPECTED:
            return "inspected";
        case Module.ActivityType.MOVED:
            return "moved";
        case Module.ActivityType.UPDATED:
            return "updated";
    }
};

interface EventIconAndColor {
    icon: IconName;
}

const eventIcon = (operation: Module.ActivityType): EventIconAndColor => {
    switch (operation) {
        case Module.ActivityType.FAVORITE:
            return {icon: "starFilled"};
        case Module.ActivityType.DOWNLOAD:
            return {icon: "download"};
        case Module.ActivityType.UPDATED:
            return {icon: "refresh"};
        case Module.ActivityType.DELETED:
            return {icon: "close"};
        case Module.ActivityType.MOVED:
            return {icon: "move"};
        default:
            return {icon: "ellipsis"};
    }
};

function groupActivity(items: Module.Activity[] = []): Module.ActivityGroup[] {
    const result: Module.ActivityGroup[] = [];
    let currentGroup: Module.ActivityGroup | null = null;

    const pushGroup = () => {
        if (currentGroup != null) {
            result.push(currentGroup);
            currentGroup = null;
        }
    };

    const initializeGroup = (item: Module.Activity) => {
        currentGroup = {
            type: item.type,
            newestTimestamp: item.timestamp,
            items: [item],
            numberOfHiddenResults: null
        };
    };

    items.forEach(item => {
        if (currentGroup === null) {
            initializeGroup(item);
        } else {
            if (currentGroup.type !== item.type ||
                Math.abs(item.timestamp - currentGroup.newestTimestamp) > (1000 * 60 * 15)) {
                pushGroup();
                initializeGroup(item);
            } else {
                currentGroup.items.push(item);
            }
        }
    });

    pushGroup();
    return result;
}

const TFRow = styled(TableRow)`
    vertical-align: top;

    & a {
        color: ${colors["text"]}
    }

    & a:hover {
        color: ${colors["textHighlight"]}
    }

    & div.ellipsis {
        white-space: nowrap;
        overflow: hidden;
        text-overflow: ellipsis;
        max-width: 100%;
        display: inline-block;
        vertical-align: bottom;
    }
`;
