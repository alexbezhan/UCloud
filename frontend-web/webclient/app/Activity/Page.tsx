import {ActivityDispatchProps, ActivityFilter, ActivityProps} from "Activity";
import * as Module from "Activity";
import {ActivityReduxObject, ReduxObject} from "DefaultObjects";
import {MainContainer} from "MainContainer/MainContainer";
import {setRefreshFunction} from "Navigation/Redux/HeaderActions";
import {setActivePage, updatePageTitle} from "Navigation/Redux/StatusActions";
import * as React from "react";
import {connect} from "react-redux";
import {Dispatch} from "redux";
import * as Scroll from "Scroll";
import {Box, InputGroup, Label} from "ui-components";
import BaseLink from "ui-components/BaseLink";
import ClickableDropdown from "ui-components/ClickableDropdown";
import {DatePicker} from "ui-components/DatePicker";
import * as Heading from "ui-components/Heading";
import {SidebarPages} from "ui-components/Sidebar";
import {ActivityFeedFrame, ActivityFeedItem, ActivityFeedSpacer} from "./Feed";
import {fetchActivity, resetActivity, setLoading, updateActivityFilter} from "./Redux/ActivityActions";
import Input from "ui-components/Input";
import {Client} from "Authentication/HttpClientInstance";

const scrollSize = 250;

const dropdownOptions: Array<{text: string; value: string}> = [
    {value: "NO_FILTER", text: "Don't filter"},
    {value: Module.ActivityType.DELETED, text: "Deletions"},
    {value: Module.ActivityType.DOWNLOAD, text: "Downloads"},
    {value: Module.ActivityType.FAVORITE, text: "Favorites"},
    {value: Module.ActivityType.MOVED, text: "Moves"},
    {value: Module.ActivityType.COPIED, text: "Copies"},
    {value: Module.ActivityType.ALLUSEDINAPP, text: "Files Used By App"},
    {value: Module.ActivityType.DIRECTORYCREATED, text: "Directory Creation"},
    {value: Module.ActivityType.RECLASSIFIED, text: "Reclassifications"},
    {value: Module.ActivityType.UPLOADED, text: "Uploads"},
    {value: Module.ActivityType.UPDATEDACL, text: "ACL Updates"},
    {value: Module.ActivityType.SHAREDWITH, text: "Shares"}
];

function Activity(props: ActivityProps): JSX.Element {

    React.useEffect(() => {
        props.onMount();
        props.resetActivity();
        props.fetchActivity({scrollSize});
        props.setRefresh(() => {
            props.resetActivity();
            props.fetchActivity({scrollSize}, props);
        });
        return () => props.setRefresh();
    }, []);

    function renderHeader(): React.ReactNode {
        return <Heading.h2>File Activity</Heading.h2>;
    }

    function renderMain(): React.ReactNode {
        const {scroll, loading, fetchActivity} = props;
        return (
            <Scroll.List
                scroll={scroll}
                scrollSize={scrollSize}
                onNextScrollRequested={req => fetchActivity(req, props)}
                loading={loading}
                frame={(ref, children) => <ActivityFeedFrame containerRef={ref}>{children}</ActivityFeedFrame>}
                renderer={props => <ActivityFeedItem activity={props.item} />}
                spacer={height => <ActivityFeedSpacer key={`spacer${height}`} height={height} />}
            />
        );
    }

    function renderSidebar(): React.ReactNode {
        const {minTimestamp, maxTimestamp, type} = props;

        const ref = React.useRef<number>(-1);

        const onKeyUp = React.useCallback((e) => {
            const user = e.target?.value ?? "";
            if (ref.current !== -1) {
                window.clearTimeout(ref.current);
            }

            ref.current = (window.setTimeout(() => {
                applyFilter({user});
            }, 500));

        }, []);

        return (
            <>
                {renderQuickFilters()}
                <Heading.h3>Active Filters</Heading.h3>
                <Box mb={16}>
                    <Label>Filter by event type</Label>
                    <ClickableDropdown
                        chevron
                        options={dropdownOptions}
                        trigger={type === undefined ? "Don't filter" : dropdownOptions.find(i => i.value === type)!.text}
                        onChange={e => applyFilter({type: e === "NO_FILTER" ? undefined : e as Module.ActivityType})}
                    />
                </Box>

                {!Client.hasActiveProject ? null :
                    <Box mb={16}>
                        <Label>Filter by username</Label>
                        <InputGroup>
                            <Input
                                placeholder={"Don't filter"}
                                onKeyUp={onKeyUp}
                            />
                        </InputGroup>
                    </Box>
                }

                <TimeFilter
                    text={"Event created after"}
                    selected={minTimestamp}
                    onChange={minTimestamp => applyFilter({minTimestamp})}
                />

                <TimeFilter
                    text={"Event created before"}
                    selected={maxTimestamp}
                    onChange={maxTimestamp => applyFilter({maxTimestamp})}
                />
            </>
        );
    }

    function renderQuickFilters(): React.ReactNode {
        const now = new Date();
        const startOfToday = getStartOfDay(now);
        const startOfWeek = getStartOfWeek(now);
        const startOfYesterday = getStartOfDay(new Date(startOfToday.getTime() - 1));

        return (
            <Box mb={16}>
                <Heading.h3>Quick Filters</Heading.h3>
                <Box mb={16}>
                    {filter("Today", {minTimestamp: startOfToday, maxTimestamp: undefined})}
                    {filter("Yesterday", {maxTimestamp: startOfToday, minTimestamp: startOfYesterday})}
                    {filter("This week", {minTimestamp: startOfWeek, maxTimestamp: undefined})}
                    {filter("No filter", {minTimestamp: undefined, maxTimestamp: undefined, type: undefined})}
                </Box>
            </Box>
        );
    }

    function filter(title: string, filter: Partial<ActivityFilter>): JSX.Element {
        return (
            <BaseLink
                style={{display: "block"}}
                href="#"
                onClick={e => {
                    e.preventDefault();
                    applyFilter(filter);
                }}
            >
                {title}
            </BaseLink>
        );
    }

    function applyFilter(filter: Partial<ActivityFilter>): void {
        props.updateFilter(filter);
        props.resetActivity();
        props.fetchActivity({scrollSize}, {...props, ...filter});
    }

    return (
        <MainContainer
            main={renderMain()}
            header={renderHeader()}
            sidebar={renderSidebar()}
            sidebarSize={340}
        />
    );
}

export const getStartOfDay = (d: Date): Date => {
    const copy = new Date(d);
    copy.setHours(0);
    copy.setMinutes(0);
    copy.setSeconds(0);
    copy.setMilliseconds(0);
    return copy;
};

export const getStartOfWeek = (d: Date): Date => {
    const day = d.getDay();
    const diff = d.getDate() - day + (day === 0 ? -6 : 1);

    const copy = new Date(d);
    copy.setDate(diff);
    copy.setHours(0);
    copy.setMinutes(0);
    copy.setSeconds(0);
    copy.setMilliseconds(0);
    return copy;
};

export const TimeFilter = (props: {text: string; onChange: (ts?: Date) => void; selected?: Date}): JSX.Element => (
    <Box mb={16}>
        <Label>{props.text}</Label>
        <InputGroup>
            <DatePicker
                showTimeInput
                placeholderText={"Don't filter"}
                selected={props.selected}
                onChange={ts => props.onChange(ts ?? undefined)}
                timeIntervals={15}
                isClearable
                selectsStart
                timeFormat="HH:mm"
                dateFormat="dd/MM/yy HH:mm"
            />
        </InputGroup>
    </Box>
);

const mapStateToProps = ({activity}: ReduxObject): ActivityReduxObject & Module.ActivityOwnProps => activity;

const mapDispatchToProps = (dispatch: Dispatch): ActivityDispatchProps => ({
    fetchActivity: async (req, filter) => {
        dispatch(setLoading(true));
        dispatch(await fetchActivity(req, filter));
    },

    onMount: () => {
        dispatch(updatePageTitle("Activity"));
        dispatch(setActivePage(SidebarPages.Activity));
    },

    resetActivity: () => {
        dispatch(resetActivity());
    },

    setRefresh: refresh => dispatch(setRefreshFunction(refresh)),

    updateFilter: (filter) => {
        dispatch(updateActivityFilter(filter));
    },
});

export default connect(mapStateToProps, mapDispatchToProps)(Activity);
