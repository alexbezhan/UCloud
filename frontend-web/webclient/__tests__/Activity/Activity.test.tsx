import * as React from "react";
import {Provider} from "react-redux";
import {create} from "react-test-renderer";
import {ThemeProvider} from "styled-components";
import Activity from "../../app/Activity/Page";
import theme from "../../app/ui-components/theme";
import {store} from "../../app/Utilities/ReduxUtilities";

test("Mount Activity Page", () => expect(create(
    <Provider store={store}>
        <ThemeProvider theme={theme}>
            <Activity />
        </ThemeProvider>
    </Provider>
)).toMatchSnapshot());
