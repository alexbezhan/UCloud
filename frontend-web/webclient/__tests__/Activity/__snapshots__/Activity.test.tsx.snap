// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mount Activity Page 1`] = `
.c1 {
  margin-left: 190px;
  padding-top: 48px;
  padding-bottom: 14px;
  padding-left: 14px;
  padding-right: 0;
  background-color: #fff;
}

.c6 {
  margin-bottom: 16px;
}

.c15 {
  padding-top: 96px;
  padding-right: 340px;
}

.c18 {
  font-smoothing: antialiased;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-family: inherit;
  font-weight: 700;
  line-height: 1.5;
  cursor: pointer;
  border-radius: 5px;
  background-color: var(--blue,#f00);
  color: var(--white,#f00);
  border-width: 0;
  border-style: solid;
  -webkit-transition: cubic-bezier(0.5,0,0.25,1) 60ms;
  transition: cubic-bezier(0.5,0,0.25,1) 60ms;
  font-size: 14px;
  padding: 9.5px 18px;
}

.c18:disabled {
  opacity: 0.25;
}

.c18:focus {
  outline: none;
}

.c18:hover {
  -webkit-transform: scale(1.03);
  -ms-transform: scale(1.03);
  transform: scale(1.03);
}

.c12 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  vertical-align: middle;
  cursor: inherit;
  margin-left: .7em;
}

.c17 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c3 {
  cursor: inherit;
  font-size: 32px;
  margin: 0;
  font-weight: 700;
}

.c7 {
  cursor: inherit;
  font-size: 24px;
  margin: 0;
  font-weight: 400;
}

.c11 {
  cursor: pointer;
}

.c9 {
  font-size: 10px;
  -webkit-letter-spacing: 0.2px;
  -moz-letter-spacing: 0.2px;
  -ms-letter-spacing: 0.2px;
  letter-spacing: 0.2px;
  display: block;
  margin: 0;
  font-size: 14px;
  color: #000;
  font-weight: 700;
  width: 100%;
}

.c10 {
  position: relative;
  display: inline-block;
}

.c10:hover > div {
  display: block;
}

.c8 {
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #1e252e;
}

.c8:hover {
  color: #006aff;
}

.c13 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 5px;
  border-width: 1px;
  border-style: solid;
  border-color: #c9d3df;
}

.c13 > .c0 {
  width: 100%;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
}

.c13 .sc-fzokOt {
  border: 0;
  box-shadow: none;
}

.c2 {
  padding-top: 14px;
  padding-bottom: 14px;
  padding-left: 204px;
  padding-right: 14px;
  width: 100%;
  height: 96px;
  background-color: #fff;
  position: absolute;
  top: 48px;
  left: 0;
  position: fixed;
  z-index: 50;
}

.c5 {
  padding-top: 144px;
  padding-left: 14px;
  padding-right: 14px;
  width: 340px;
  height: 100%;
  position: absolute;
  top: 0;
  right: 0;
  overflow-y: auto;
  position: fixed;
  z-index: 45;
}

.c14 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  display: block;
  width: 100%;
  font-family: inherit;
  color: inherit;
  font-size: 14px;
  background-color: transparent;
  border-radius: 5px;
  border-width: 0px;
  border-style: solid;
  border-color: var(--borderGray,#f00);
  padding-top: 14px;
  padding-bottom: 14px;
  padding-left: 12px;
  padding-right: 12px;
  margin: 0;
}

.c14::-webkit-input-placeholder {
  color: var(--gray,#f00);
}

.c14::-moz-placeholder {
  color: var(--gray,#f00);
}

.c14:-ms-input-placeholder {
  color: var(--gray,#f00);
}

.c14::placeholder {
  color: var(--gray,#f00);
}

.c14::-ms-clear {
  display: none;
}

@media screen and (max-width:511px) {
  .c4 {
    display: none;
  }
}

@media screen and (min-width:512px) and (max-width:639px) {
  .c4 {
    display: none;
  }
}

@media screen and (min-width:640px) and (max-width:767px) {
  .c4 {
    display: none;
  }
}

@media screen and (min-width:768px) and (max-width:1023px) {
  .c16 {
    display: none;
  }
}

@media screen and (min-width:1024px) and (max-width:1279px) {
  .c16 {
    display: none;
  }
}

@media screen and (min-width:1280px) {
  .c16 {
    display: none;
  }
}

<div
  className="c0 c1"
>
  <div
    className="c0 sc-AxiKw c2"
    height={96}
    width={1}
  >
    <h2
      className="c3"
      cursor="inherit"
      fontSize={5}
    >
      File Activity
    </h2>
  </div>
  <div
    className="c0 c4"
  >
    <div
      className="c0 sc-AxiKw c5"
      height="100%"
      width={340}
    >
      <div
        className="c0 c6"
      >
        <h3
          className="c7"
          cursor="inherit"
          fontSize={4}
        >
          Quick Filters
        </h3>
        <div
          className="c0 c6"
        >
          <a
            className="c8"
            color="text"
            href="#"
            onClick={[Function]}
            style={
              Object {
                "display": "block",
              }
            }
          >
            Today
          </a>
          <a
            className="c8"
            color="text"
            href="#"
            onClick={[Function]}
            style={
              Object {
                "display": "block",
              }
            }
          >
            Yesterday
          </a>
          <a
            className="c8"
            color="text"
            href="#"
            onClick={[Function]}
            style={
              Object {
                "display": "block",
              }
            }
          >
            This week
          </a>
          <a
            className="c8"
            color="text"
            href="#"
            onClick={[Function]}
            style={
              Object {
                "display": "block",
              }
            }
          >
            No filter
          </a>
        </div>
      </div>
      <h3
        className="c7"
        cursor="inherit"
        fontSize={4}
      >
        Active Filters
      </h3>
      <div
        className="c0 c6"
      >
        <label
          className="c9"
          color="black"
          fontSize={1}
          fontWeight="bold"
          width="100%"
        >
          Filter by event type
        </label>
        <div
          className="c10"
          data-tag="dropdown"
        >
          <span
            className="c11"
            cursor="pointer"
            onClick={[Function]}
          >
            Don't filter
            <svg
              className="c12"
              clipRule="evenodd"
              cursor="inherit"
              fill="currentcolor"
              fillRule="evenodd"
              height=".7em"
              ml=".7em"
              viewBox="0 0 24 20"
              width=".7em"
            >
              <path
                d="M11.998 19.999c-.598 0-1.028-.61-1.028-.61L.365 2.268C-.453 1.02.153 0 1.703 0h20.593c1.554 0 2.156 1.02 1.34 2.27l-10.61 17.121c0-.002-.43.608-1.028.608z"
                fillRule="nonzero"
              />
            </svg>
          </span>
        </div>
      </div>
      <div
        className="c0 c6"
      >
        <label
          className="c9"
          color="black"
          fontSize={1}
          fontWeight="bold"
          width="100%"
        >
          Event created after
        </label>
        <div
          className="c13"
        >
          <div
            className="react-datepicker-wrapper"
          >
            <div
              className="react-datepicker__input-container"
            >
              <input
                className="c14"
                disabled={false}
                onBlur={[Function]}
                onChange={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                placeholder="Don't filter"
                readOnly={false}
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="c0 c6"
      >
        <label
          className="c9"
          color="black"
          fontSize={1}
          fontWeight="bold"
          width="100%"
        >
          Event created before
        </label>
        <div
          className="c13"
        >
          <div
            className="react-datepicker-wrapper"
          >
            <div
              className="react-datepicker__input-container"
            >
              <input
                className="c14"
                disabled={false}
                onBlur={[Function]}
                onChange={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                placeholder="Don't filter"
                readOnly={false}
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="c0 c15"
  >
    <div
      className="c0 c16"
    >
      <div
        className="c0 c6"
      >
        <h3
          className="c7"
          cursor="inherit"
          fontSize={4}
        >
          Quick Filters
        </h3>
        <div
          className="c0 c6"
        >
          <a
            className="c8"
            color="text"
            href="#"
            onClick={[Function]}
            style={
              Object {
                "display": "block",
              }
            }
          >
            Today
          </a>
          <a
            className="c8"
            color="text"
            href="#"
            onClick={[Function]}
            style={
              Object {
                "display": "block",
              }
            }
          >
            Yesterday
          </a>
          <a
            className="c8"
            color="text"
            href="#"
            onClick={[Function]}
            style={
              Object {
                "display": "block",
              }
            }
          >
            This week
          </a>
          <a
            className="c8"
            color="text"
            href="#"
            onClick={[Function]}
            style={
              Object {
                "display": "block",
              }
            }
          >
            No filter
          </a>
        </div>
      </div>
      <h3
        className="c7"
        cursor="inherit"
        fontSize={4}
      >
        Active Filters
      </h3>
      <div
        className="c0 c6"
      >
        <label
          className="c9"
          color="black"
          fontSize={1}
          fontWeight="bold"
          width="100%"
        >
          Filter by event type
        </label>
        <div
          className="c10"
          data-tag="dropdown"
        >
          <span
            className="c11"
            cursor="pointer"
            onClick={[Function]}
          >
            Don't filter
            <svg
              className="c12"
              clipRule="evenodd"
              cursor="inherit"
              fill="currentcolor"
              fillRule="evenodd"
              height=".7em"
              ml=".7em"
              viewBox="0 0 24 20"
              width=".7em"
            >
              <path
                d="M11.998 19.999c-.598 0-1.028-.61-1.028-.61L.365 2.268C-.453 1.02.153 0 1.703 0h20.593c1.554 0 2.156 1.02 1.34 2.27l-10.61 17.121c0-.002-.43.608-1.028.608z"
                fillRule="nonzero"
              />
            </svg>
          </span>
        </div>
      </div>
      <div
        className="c0 c6"
      >
        <label
          className="c9"
          color="black"
          fontSize={1}
          fontWeight="bold"
          width="100%"
        >
          Event created after
        </label>
        <div
          className="c13"
        >
          <div
            className="react-datepicker-wrapper"
          >
            <div
              className="react-datepicker__input-container"
            >
              <input
                className="c14"
                disabled={false}
                onBlur={[Function]}
                onChange={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                placeholder="Don't filter"
                readOnly={false}
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="c0 c6"
      >
        <label
          className="c9"
          color="black"
          fontSize={1}
          fontWeight="bold"
          width="100%"
        >
          Event created before
        </label>
        <div
          className="c13"
        >
          <div
            className="react-datepicker-wrapper"
          >
            <div
              className="react-datepicker__input-container"
            >
              <input
                className="c14"
                disabled={false}
                onBlur={[Function]}
                onChange={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                placeholder="Don't filter"
                readOnly={false}
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div>
      <h2
        className="c3"
        cursor="inherit"
        fontSize={5}
      >
        No results.
      </h2>
    </div>
    <div
      className="c17"
    >
      <button
        className="c18"
        color="blue"
        disabled={false}
        onClick={[Function]}
      >
        Load more
      </button>
    </div>
  </div>
</div>
`;
